tools/triage/utils/extraction/jira_attachment_querier/jira_remote_archive_querier.py

```diff
@@ -0,0 +1,242 @@
+import logging
+import shutil
+import tempfile
+from pathlib import Path
+from typing import Optional, final
+
+import jira
+import nestedarchive
+
+from tools.triage.utils.extraction.jira_attachment_querier.exceptions import (
+    FailedToExtractAttachment,
+    FailedToGetAttachmentInIssue,
+)
+from tools.triage.utils.extraction.jira_attachment_querier.local_nested_archive_extractor import (
+    LocalNestedArchiveExtractor,
+)
+
```

Comment:
I created two options for getting the target file -
1. Extracting the whole archive using `LocalNestedArchiveExtractor` (for debugging purposes mainly as we can see the whole tree).
2. Extracting only necessary archives using `nestedarchive.get` .